core/src/main/java/bio/terra/pearl/core/service/survey/SurveyParseUtils.java

```diff
@@ -131,6 +131,26 @@ public static SurveyQuestionDefinition unmarshalSurveyQuestion(
         return definition;
     }
 
+    private static Pattern surveyQuestionRegexPattern = Pattern.compile("\\{\\s*\\w+\\.[\\w.]+\\s*}");
+
+    public static Map<String, List<String>> findReferencedSurveyQuestions(Survey survey) {
+        return surveyQuestionRegexPattern.matcher(survey.getContent()).results().map(result -> {
+            String cleaned = result.group().substring(1, result.group().length() - 1).trim();
+
+            int periodIndex = cleaned.indexOf('.');
+
+            String surveyStableId = cleaned.substring(0, periodIndex);
+            String questionStableId = cleaned.substring(periodIndex + 1);
```

Comment:
let's use `split(".")` to make it clearer what's going on and to avoid off-by-ones

Reply:
It's definitely better, but I'm not sure it's much cleaner (the issue being that, e.g. `survey1.matrixQuestion.specificQuestion` is possible, so we have to account for multiple periods in question names)